site/app/entities/forum/Thread.php

```diff
@@ -157,6 +188,18 @@ public function getPosts(): Collection {
     public function getCategories(): Collection {
         return $this->categories;
     }
+
+    /**
+     * @param Category[]|Collection<Category> $categories
+     * @return void
+     */
+    public function setCategories(array|Collection $categories): void {
```

Comment:
Is input type an array or collection? Doesn't make sense to support both typing wise.